lib/app/server.js

```diff
@@ -153,11 +153,16 @@ export default async ssrContext => {
   ** Call .validate()
   */
   let isValid = true
-  Components.forEach((Component) => {
+  let validPromises = await Promise.all(Components.map((Component) => {
```

Comment:
The `Components` even if there is no `validate` in it will become a fast return promise, it should be duplicate, maybe you can filter out them.